include/deal.II/grid/filtered_iterator.h

```diff
@@ -960,10 +960,8 @@ FilteredIterator<BaseIterator> &
 FilteredIterator<BaseIterator>::
 operator = (const FilteredIterator &fi)
 {
-  Assert ((fi.state() != IteratorState::valid) || (*predicate)(fi),
-          ExcInvalidElement(fi));
-  BaseIterator::operator = (fi);
-  return *this;
+  const BaseIterator &bi = fi;
+  return operator = (bi);
```

Comment:
Ah, I missed that you already have a failing testcase (which, incidentally, doesn't seem to fail for me, so it must be a problem with the compiler).

If it's easy to extract a small self-contained testcase, that would be appreciated. If not, let's not worry about it.